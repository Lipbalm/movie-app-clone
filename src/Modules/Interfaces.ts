export interface ITopRatedMovies<T> {
  page: number;
  total_results: number;
  total_pages: number;
  results: T;
}

export interface ITopRatedResults {
  popularity: number;
  vote_count: number;
  video: boolean;
  poster_path: string;
  id: number;
  adult: false;
  backdrop_path: string;
  original_language: string;
  original_title: string;
  genre_ids: number[];
  vote_average: number;
  overview: string;
  release_date: string;
  title: string;
}

export interface IGenreIds {
  id: number;
  name: string;
}
