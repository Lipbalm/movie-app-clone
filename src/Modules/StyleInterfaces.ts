export interface IStyleProperty {
  [key: string]: string;
}
export interface ICSSPesudo {
  [key: string]: IStyleProperty;
}
export interface IStyleFrame {
  basis: IStyleProperty;
  pesudo?: ICSSPesudo;
}
export interface IStyle {
  styleObj?: IStyleFrame;
}
