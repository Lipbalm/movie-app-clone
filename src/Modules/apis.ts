import axios, { AxiosInstance } from "axios";
import {
  ITopRatedMovies,
  ITopRatedResults,
  IGenreIds,
  IMapGenreIds
} from "./Interfaces";

const apiKey = "f14b5a458d4b24203b40ecec534dfd0d";
const baseMoviesInstance: AxiosInstance = axios.create({
  baseURL: `https://api.themoviedb.org/3/movie`
});

const imageBaseURL: string = "https://image.tmdb.org/t/p/w500/";

let genreIds: IMapGenreIds[];

const getTopRatedMovie = (language: string, page: number) =>
  baseMoviesInstance
    .get<ITopRatedMovies<ITopRatedResults[]>>(
      `/top_rated?api_key=${apiKey}&language=${language}&page=${page}`
    )
    .then(response => response.data);

const getGenreIds = (language: string) =>
  axios
    .get<IGenreIds[]>(
      `https://api.themoviedb.org/3/genre/movie/list?api_key=${apiKey}&language=${language}`
    )
    .then(response => {
      response.data.forEach(v =>
        v.name === undefined ? (genreIds[v.id] = "") : (genreIds[v.id] = v.name)
      );
      return true;
    })
    .catch(e => e);

export { getTopRatedMovie, genreIds, getGenreIds, imageBaseURL };
