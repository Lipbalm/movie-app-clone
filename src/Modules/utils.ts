function isMatch(input: string, pattern: string): boolean {
  let regex = new RegExp(pattern, "g");
  return regex.test(input);
}
export { isMatch };
