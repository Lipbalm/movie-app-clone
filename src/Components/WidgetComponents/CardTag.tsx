import React, { FC } from "react";
import { BottomTagWrapper, BottomGerenTag, ItemTitle } from "./WidgetStyle";
import { IStyleFrame } from "../../Modules/StyleInterfaces";

interface IProps {
  genres?: string;
  title: string;
  type?: string;
}

const itemTitleStyle: IStyleFrame = {
  basis: {
    "font-size": "1.15rem",
    "font-weight": "600",
    "color": "black"
  }
};

export const BottomTag: FC<IProps> = ({ genres, title }) => {
  return (
    <BottomTagWrapper>
      <BottomGerenTag data-testid="genre">
        {genres === undefined || genres}
      </BottomGerenTag>
      <ItemTitle data-testid="title" styleObj={itemTitleStyle}>
        {title}
      </ItemTitle>
    </BottomTagWrapper>
  );
};
