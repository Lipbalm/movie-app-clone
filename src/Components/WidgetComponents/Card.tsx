import React, { FC } from "react";
import { ITopRatedResults } from "../../Modules/Interfaces";
import { RectCard, ThumnailWrapper, ThumnailImg } from "../../Style";
import { baseImageURL } from "../../Modules/apis";

interface IProps {
  movie: ITopRatedResults;
  index: number;
}

const Card: FC<IProps> = ({ movie, index }) => {
  const {
    id,
    backdrop_path,
    genre_ids,
    adult,
    original_title,
    poster_path,
    vote_average
  } = movie;
  return (
    <RectCard isMargin={index > 0 ? true : false}>
      <ThumnailWrapper>
        <ThumnailImg src={`${baseImageURL}${poster_path}`} alt="no Images" />
      </ThumnailWrapper>
    </RectCard>
  );
};

export default Card;
