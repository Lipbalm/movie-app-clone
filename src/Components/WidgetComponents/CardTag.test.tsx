import React from "react";
import { render } from "@testing-library/react";
import { BottomTag, OverlayTag } from "./CardTag";
import { tempCard } from "../../Test/TempData/Card";
import { getGenreIds } from "../../Modules/apis";
//import { changeGenreIdToName } from "../../Modules/utils";

const { genre_ids, title } = tempCard;

const cardTagSetup = () => {
  return {
    ...render(<BottomTag genres={genre_ids.join("/")} title={title} />)
  };
};

const overlayTagSetup = () => {
  return { ...render(<OverlayTag title={title} />) };
};

describe("<CardTag /> & <OverlayTag />", () => {
  it("SnapShot", () => {
    const { container } = cardTagSetup();
    expect(container).toMatchSnapshot();
  });

  it("CardTag 장르 및 이름 보이기", () => {
    const { getByTestId } = cardTagSetup();
    const genre = getByTestId("genre");
    const name = getByTestId("title");
    expect(genre).toHaveTextContent(genre_ids.join("/"));
    expect(name).toHaveTextContent(title);
  });

  it("OverlayTag", () => {
    const { container } = overlayTagSetup();
    expect(container).toMatchSnapshot();
  });
});
