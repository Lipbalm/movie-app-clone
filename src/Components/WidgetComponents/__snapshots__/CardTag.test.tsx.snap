// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardTag /> & <OverlayTag /> OverlayTag 1`] = `
<div>
  <div
    class="sc-gqjmRU cfHsSd"
  >
    <a
      class="sc-fjdhpX bRnXaD"
      data-testid="title"
    >
      Dilwale Dulhania Le Jayenge
    </a>
  </div>
</div>
`;

exports[`<CardTag /> & <OverlayTag /> SnapShot 1`] = `
<div>
  <div
    class="sc-gZMcBi cUITsv"
  >
    <span
      class="sc-VigVT jnzVBN"
      data-testid="genre"
    >
      35/18/10749
    </span>
    <a
      class="sc-jTzLTM djLRRX"
      data-testid="title"
    >
      Dilwale Dulhania Le Jayenge
    </a>
  </div>
</div>
`;
